static/elements/chromedash-approvals-dialog.js

```diff
@@ -1,42 +1,368 @@
       canApprove: {type: Boolean},
+      feature: {type: Object},
+      approvals: {type: Array},
+      comments: {type: Array},
+      showAllIntents: {type: Boolean},
+      changedApprovalsByField: {attribute: false},
+      needsSave: {type: Boolean},
     };
   }
 
   constructor() {
     super();
     this.signedInUser = ''; // email address
     this.canApprove = false;
+    this.featureId = 0;
+    this.feature = {};
+    this.approvals = [];
+    this.comments = [];
+    this.subsetPending = false;
+    this.showAllIntents = false;
+    this.changedApprovalsByField = new Map();
+    this.needsSave = false;
   }
 
   static get styles() {
     return [
       SHARED_STYLES,
       css`
+        h3 {
+          margin: var(--content-padding-half);
+        }
+
+        .approval_section {
+          margin-top: var(--content-padding);
+        }
+
+        .comment_section {
+          max-height: 250px;
+          overflow-y: scroll;
+        }
+
+        .approval_section div,
+        .comment {
+          margin-left: var(--content-padding);
+        }
+
+        .approval_row {
+          width: 650px;
+          margin-bottom: var(--content-padding-half);
+        }
+
+        .set_by,
+        .set_on,
+        .appr_val {
+          display: inline-block;
+          width: 200px;
+          margin-right: var(--content-padding-half);
+          overflow: hidden;
+          text-overflow: ellipsis;
+          white-space: nowrap;
+        }
+
+        select {
+          margin: 0;
+        }
+
+        .comment_body {
+          background: var(--table-alternate-background);
+          padding: var(--content-padding-half);
+          white-space: pre-wrap;
+          width: 46em;
+          margin-bottom: var(--content-padding);
+        }
+
+        .controls {
+          padding: var(--content-padding);
+          text-align: right;
+        }
+
+        #show_all_checkbox {
+         float: left;
+        }
+
+        textarea {
+          padding: 4px;
+        }
       `];
   }
 
   openWithFeature(featureId) {
     this.featureId = featureId;
-    this.shadowRoot.querySelector('chromedash-dialog').open();
+    const p1 = window.csClient.getFeature(this.featureId).then(
+      (feature) => {
+        this.feature = feature;
+      });
+    const p2 = window.csClient.getApprovals(this.featureId).then(
+      (res) => {
+        this.approvals = res.approvals;
+        const numPending = this.approvals.filter((av) =>
+          ACTIVE_STATES.includes(av.state)).length;
+        this.subsetPending = (numPending > 0 &&
+                              numPending < APPROVAL_DEFS.length);
+        this.showAllIntents = numPending == 0;
+        this.changedApprovalsByField = new Map();
+        this.needsSave = false;
+      });
+    const p3 = window.csClient.getComments(this.featureId).then(
+      (res) => {
+        this.comments = res.comments;
+      });
+    Promise.all([p1, p2, p3]).then(() => {
```

Comment:
Now that this part is async, do we need to show messages like "Loading...", and catch any failed promises and show some error message?
(if the service is reasonably reliable and fast, then maybe it's fine without thoes)

Reply:
Added.